service/src/main/java/bio/terra/pipelines/dependencies/rawls/RawlsService.java

```diff
@@ -108,11 +120,150 @@ public Entity getDataTableEntity(
                 .getEntity(workspaceNamespace, workspaceName, entityType, entityName, null, null));
   }
 
+  public MethodConfiguration getCurrentMethodConfigForMethod(
+      String accessToken, String workspaceNamespace, String workspaceName, String methodName) {
+    return executionWithRetryTemplate(
+        listenerResetRetryTemplate,
+        () ->
+            rawlsClient
+                .getMethodConfigsApi(accessToken)
+                .getMethodConfiguration(
+                    workspaceNamespace, workspaceName, workspaceNamespace, methodName));
+  }
+
+  public ValidatedMethodConfiguration setMethodConfigForMethod(
+      String accessToken,
+      MethodConfiguration methodConfiguration,
+      String workspaceNamespace,
+      String workspaceName,
+      String methodName) {
+    return executionWithRetryTemplate(
+        listenerResetRetryTemplate,
+        () ->
+            rawlsClient
+                .getMethodConfigsApi(accessToken)
+                .updateMethodConfiguration(
+                    methodConfiguration,
+                    workspaceNamespace,
+                    workspaceName,
+                    workspaceNamespace,
+                    methodName));
+  }
+
   // returns true if submission is in a running state
   public static boolean submissionIsRunning(Submission submission) {
     return !FINAL_RUN_STATES.contains(submission.getStatus());
   }
 
+  /**
+   * validates a method config against the expected version, inputs, and outputs it does this by
+   * checking that the methodRepoMethod method version matches we expect and that all expected
```

Comment:
- "methodRepoMethod method version" can just be "wdlMethodVersion" 
- "matches **what** we expect"
- (next line) "inputs outputs both exist" -> "inputs/outputs exist"